crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -393,6 +394,26 @@ impl Query {
             .await
             .extend()
     }
+
+    /// Verify a zkLogin signature.
+    async fn verify_zklogin_signature(
+        &self,
+        ctx: &Context<'_>,
+        bytes: String,
+        signature: String,
+        intent_scope: u64,
```

Comment:
you are correct that there are 2 possible values right now 0 or 3. im not sure if we plan to add new ones, if i put it in an enum, is it ok to change in the future?

this is the native definition using https://doc.rust-lang.org/nomicon/repr-rust.html

```
#[derive(Serialize_repr, Deserialize_repr, Copy, Clone, PartialEq, Eq, Debug, Hash)]
#[repr(u8)]
pub enum IntentScope {
    TransactionData = 0,         // Used for a user signature on a transaction data.
    TransactionEffects = 1,      // Used for an authority signature on transaction effects.
    CheckpointSummary = 2,       // Used for an authority signature on a checkpoint summary.
    PersonalMessage = 3,         // Used for a user signature on a personal message.
    SenderSignedTransaction = 4, // Used for an authority signature on a user signed transaction.
    ProofOfPossession = 5, // Used as a signature representing an authority's proof of possession of its authority protocol key.
    HeaderDigest = 6,      // Used for narwhal authority signature on header digest.
    BridgeEventUnused = 7, // for bridge purposes but it's currently not included in messages.
    ConsensusBlock = 8,    // Used for consensus authority signature on block's digest
}
```

Reply:
also does graphql enum support integers? i looked at some examples it doesnt seem so? i want it to be consistent with the legacy zklogin verifier that this value can be either 0 or 3.

Reply:
I don't think graphql enum supports integers, we would have to do some mapping between the graphql enum (something like)
```
#[derive(enum)]
IntentScope {
  TransactionData,
  PersonalMessage
}
```
and the IntentScope from the zklogin verifier. I think to the end user this would be more interpretable than passing in an integer though?

We have a similar scenario for `ObjectStatus` https://github.com/MystenLabs/sui/blob/wlmyng/gql/simple-benchmark/crates/sui-graphql-rpc/src/types/object.rs#L76

Reply:
revised to use enum